documents/api/block/stream/block.md

```diff
@@ -0,0 +1,64 @@
+## Table of Contents
+
+- [stream/block.proto](#stream_block-proto)
+    - [Block](#com-hedera-hapi-block-stream-Block)
+  
+
+
+
+<a name="stream_block-proto"></a>
+<p align="right"><a href="#top">Top</a></p>
+
+## stream/block.proto
+# Block Stream
+The base element of the block stream _at rest_.
+A `Block` contains a record of all transactions, results, and outputs for
+a block in the chain. Each `Block` also contains a state proof for
+validation and a header with version and algorithm information.
+
+Block entries are not designed for streaming, but for storing blocks in
+persistent storage, verifying block stream data, and as query responses
+when a block is requested from a block node.
+
+### Keywords
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+"SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this
+document are to be interpreted as described in
+[RFC2119](https://www.ietf.org/rfc/rfc2119) and clarified in
+[RFC8174](https://www.ietf.org/rfc/rfc8174).
+
+
+<a name="com-hedera-hapi-block-stream-Block"></a>
+
+### Block
+A single complete Hedera block chain block.
+
+This is a single block structure and SHALL NOT represent the primary
```

Comment:
Is the goal here saying `BlockItems` are streamed not `Block` objects. 
So to a client streaming they wouldn't know they have a complete block without the `BlockHeader` and `BlockStateProof` delineators?

Is the reasoning size or something else?

The impact is clients need to keep track of the separator to create a meaningful Block aggregation

Reply:
Absolutely.
The stream is individual block items, not entire blocks.
Jasper has a good description for why we chose that approach, but it comes down to efficiency and resilience, for the most part.

A client that is streaming will only handle full blocks if the client chooses to do so, but if it does it will require both header and proof to delimit the block.

Clients that handle full blocks need to handle validating the block proof and may need to maintain an entire mirror state as well (if it's a stateful node), so tracking when a block starts and ends isn't hard in context.

Reply:
Note, for "casual" clients, who only want full blocks, the expectation is that the client will just request a single block at a time from a block node, rather than requesting a stream from the block node.
(We expect _push_ streams to be restricted to specific consensus nodes sending to specific block nodes, at least initially).